Create a quantized in-palce version CUDA ReLU function, relu_quantized_cuda_.

fufeisi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85670
Summary:
this and #85669 are to allow the relu function to run on a quantized tensor on cuda. That is torch.relu(qa) for a quantized tensor qa on cuda.
Test Plan:
python test/test_quantization.py

Previous PR that has been reverted: #85502.

fufeisi committed:
Create a quantized in-palce version CUDA ReLU function, relu_quantized_cuda_.

[ghstack-poisoned]

fufeisi committed:
Update on "Create a quantized in-palce version CUDA ReLU function, relu_quantized_cuda_."




[ghstack-poisoned]

fufeisi committed:
Update on "Create a quantized in-palce version CUDA ReLU function, relu_quantized_cuda_."

Summary:
this and #85669 are to allow the relu function to run on a quantized tensor on cuda. That is torch.relu(qa) for a quantized tensor qa on cuda.
Test Plan:
python test/test_quantization.py


[ghstack-poisoned]

fufeisi committed:
Update on "Create a quantized in-palce version CUDA ReLU function, relu_quantized_cuda_."

Summary:
this and #85669 are to allow the relu function to run on a quantized tensor on cuda. That is torch.relu(qa) for a quantized tensor qa on cuda.
Test Plan:
python test/test_quantization.py


[ghstack-poisoned]

fufeisi committed:
Update on "Create a quantized in-palce version CUDA ReLU function, relu_quantized_cuda_."

Summary:
this and #85669 are to allow the relu function to run on a quantized tensor on cuda. That is torch.relu(qa) for a quantized tensor qa on cuda.
Test Plan:
python test/test_quantization.py


[ghstack-poisoned]

fufeisi committed:
Update on "Create a quantized in-palce version CUDA ReLU function, relu_quantized_cuda_."

Summary:
this and #85669 are to allow the relu function to run on a quantized tensor on cuda. That is torch.relu(qa) for a quantized tensor qa on cuda.
Test Plan:
python test/test_quantization.py


[ghstack-poisoned]

dzdang commented:
can you link the previous PR (the one with the review history) in the summary

dzdang commented on test/quantization/core/test_quantized_op.py:
```diff
@@ -255,18 +255,6 @@ def test_qrelu(self):
         ]
         devices = ["cpu", "cuda"] if TEST_CUDA else ["cpu"]
         for device in devices:
-            # Only test the non-in-place version relu quantized cuda,
```
thought I left a review comment here but can't find it. I'm wondering why we add this in the previous PR and remove it here?

fufeisi replied:
For the previous PR, we only have non-in-palce qrelu, so I add this for only test non-in-palce qrelu, avoid in-palce qrelu.

fufeisi committed:
Update on "Create a quantized in-palce version CUDA ReLU function, relu_quantized_cuda_."

Summary:
this and #85669 are to allow the relu function to run on a quantized tensor on cuda. That is torch.relu(qa) for a quantized tensor qa on cuda.
Test Plan:
python test/test_quantization.py

Previous PR that has been reverted: #85502. 

[ghstack-poisoned]

fufeisi committed:
Update on "Create a quantized in-palce version CUDA ReLU function, relu_quantized_cuda_."

Summary:
this and #85669 are to allow the relu function to run on a quantized tensor on cuda. That is torch.relu(qa) for a quantized tensor qa on cuda.
Test Plan:
python test/test_quantization.py

Previous PR that has been reverted: #85502. 

[ghstack-poisoned]

fufeisi commented:
@mergebot rebase

fufeisi commented:
@pytorchbot rebase

fufeisi committed:
Update on "Create a quantized in-palce version CUDA ReLU function, relu_quantized_cuda_."

Summary:
this and #85669 are to allow the relu function to run on a quantized tensor on cuda. That is torch.relu(qa) for a quantized tensor qa on cuda.
Test Plan:
python test/test_quantization.py

Previous PR that has been reverted: #85502. 

[ghstack-poisoned]

fufeisi commented:
@pytorchbot merge
